### PR TITLE
[shell] move common shell commands to core library

### PR DESCRIPTION
**Pull request by gjc13**
#### Problem

We'd like to have a user-friendly CLI in the devices for developing and testing as we have in OpenThread.

 #### Summary of Changes

This is the first PR of the effort to add shell as an optional component for all devices. 
Main changes include:

* Move wifi, ble and device config command to common library
* The `wifi` commands has been simplified with less modes.
  Platform-specific commands have been removed.
* The `btp` command is renamed to `ble` with all the not-implemented commands removed.
* The `device config` command is renamed to `config` with unused configs removed.
* Fix shell formatting on ESP32.

### CODE REVIEW

**Review by andy31415 — COMMENTED** [review]
Please rename code to NOT use names like helper/support/util/tools/core.

These names do not describe what the contents of the file generally is and end up being filled  with unrelated parts of code as a result. 

shell/shell_core.h stutters and if I remove the stutter I get `shell/core.h` which is unclear about what that is.

File: examples/lock-app/esp32/.gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+*.vscode
```
Comment (andy31415) [inline examples/lock-app/esp32/.gitignore:1]: I think we have a top-level gitignore. do we need one here as well?

File: examples/shell/esp32/main/main.cpp [hunk 2]
```diff
@@ -25,14 +25,42 @@
...
 
+struct ShellLineArgs
+{
+    char * line;
+    TaskHandle_t source_task;
+};
+
+static void process_shell_line(intptr_t context)
+{
+    ShellLineArgs * args = reinterpret_cast<ShellLineArgs *>(context);
```
Comment (andy31415) [inline examples/shell/esp32/main/main.cpp:43]: can we somehow make this typesafe? 

Could we make ShellLineArgs have a NotifyDone/WaitForDone method to not have the hardcoded xTaskNotify inside the members? I would find that easier to read.

File: src/lib/shell/commands/CommandBase64.cpp [hunk 3]
```diff
@@ -0,0 +1,103 @@
+/*
```
Comment (andy31415) [inline src/lib/shell/commands/CommandBase64.cpp:1]: `commands/Command` stutters. Rename to `commands/Base64.h`

File: src/lib/shell/commands/CommandBase64.cpp [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+#include <lib/support/CHIPArgParser.hpp>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/RandUtils.h>
+
+chip::Shell::Shell sShellBase64Commands;
+
+namespace chip {
+namespace Shell {
+
+int CommandBase64Help(int argc, char ** argv)
```
Comment (andy31415) [inline src/lib/shell/commands/CommandBase64.cpp:38]: should Command be a namespace? Otherwise we have all these methods smurf-named as 'Command*'

**Review by andy31415 — COMMENTED** [review]
Would also avoid 'common' in names.

**Review by jmartinez-silabs — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

File: examples/shell/esp32/main/main.cpp (hunk 2 above)
Comment (gjc13) [reply examples/shell/esp32/main/main.cpp:43]: The reinterpret cast is a must here.

**Comment by gjc13** [comment]
@andy31415 PTAL

**Review by Damian-Nordic — CHANGES_REQUESTED** [review]
Note that we had renamed `shell/shell.h` to `shell/shell_core.h` to fix filename collision with Zephyr-supplied `shell/shell.h`. Now that the file was renamed to `Shell.h` the issue is gone on Linux/mac, but may still appear on case-insensitive systems like Windows. Can we possibly rename it to something like `Engine.h`? @andy31415 would you approve such a name? :)

**Comment by gjc13** [comment]
@Damian-Nordic renamed.

**Review by Damian-Nordic — APPROVED** [review]
Thank you!

**Comment by andy31415** [comment]
> Note that we had renamed `shell/shell.h` to `shell/shell_core.h` to fix filename collision with Zephyr-supplied `shell/shell.h`. Now that the file was renamed to `Shell.h` the issue is gone on Linux/mac, but may still appear on case-insensitive systems like Windows. Can we possibly rename it to something like `Engine.h`? @andy31415 would you approve such a name? :)

Sure. Also my request to avoid odd names is not a ban, just trying hard. If there is a naming clash then that seems to warrant a name change. Would not block PR over this for sure.

**Review by andy31415 — APPROVED** [review]

**Pull request merged**